autocorpus/table.py

```diff
@@ -6,565 +6,583 @@
-                    value = re.sub(
-                        r"(\s{0,1})[*××xX](\s{0,1})10(_{0,1})", "e", value
-                    ).replace("−", "-")
-                if re.match(self.pval_scientific_regex, value):
-                    value = re.sub(r"(\s{0,1})[–−-](\s{0,1})", "-", value)
-                    value = re.sub(r"(\s{0,1})[eE]", "e", value)
-                for drow, dcol in product(range(rowspan), range(colspan)):
-                    try:
-                        table[row_idx + drow][col_idx + dcol] = value
-                        rowspans[col_idx + dcol] = rowspan
-                    except IndexError:
-                        # rowspan or colspan outside the confines of the table
-                        pass
-
-            # update rowspan bookkeeping
-            rowspans = {c: s - 1 for c, s in rowspans.items() if s > 1}
-        return table
-
-    def __check_superrow(self, row):
-        """Check if the current row is a superrow.
-
-        Args:
-            row (list): list of cells
-
-        Return:
-            True/False
-
-        """
-        cleaned_row = set(
-            i for i in row if (str(i) != "") & (str(i) != "\n") & (str(i) != "None")
-        )
-        return len(cleaned_row) == 1 and bool(
-            re.match("[a-zA-Z]", next(iter(cleaned_row)))
-        )
+    """
+    try:
+        float(s.replace(",", ""))
+        return True
+    except ValueError:
+        return False
 
-    def __get_headers(self, t, config):
-        """Identify headers from a table.
 
-        Args:
-            t (bs4.BeautifulSoup): BeautifulSoup object of table
-            config (dict): configuration dictionary
+def __is_mix(s: str) -> bool:
+    """Check if input string is a mix of number and text.
 
-        Returns:
-            (list): a list of header index
+    Args:
+        s (str): input string
 
-        Raises:
-            KeyError: Raises an exception.
+    Returns:
+        (bool): True/False
 
-        """
-        idx_list = []
-        for idx, row in enumerate(
-            get_data_element_node(config["tables"]["data"]["table-row"], t)
-        ):
-            if get_data_element_node(config["tables"]["data"]["header-element"], row):
-                idx_list.append(idx)
-            elif "class" in row.attrs:
-                if "thead" in row.attrs["class"]:
-                    idx_list.append(idx)
-        # if no table headers found
-        if idx_list == []:
-            idx_list = [0]
-        return idx_list
-
-    def __is_number(self, s):
-        """Check if input string is a number.
-
-        Args:
-            s (str): input string
-
-        Returns:
-            (bool): True/False
-
-        """
-        try:
-            float(s.replace(",", ""))
+    """
+    if any(char.isdigit() for char in s):
+        if any(char for char in s if not char.isdigit()):
             return True
-        except ValueError:
-            return False
+    return False
```

Comment:
I think this still is wrong. The second if statement just gets the chars which are not digits and checks whether any are truthy. It should check whether any chars are non-digits. (In practice it will work but it's still not quite right)

GitHub won't let me make this as a suggestion, but this would work:

```py
    if any(char.isdigit() for char in s):
        return any(not char.isdigit() for char in s)
    return False
```